src/file.cpp

```diff
@@ -1059,33 +1094,32 @@ namespace libtorrent
 		ec.clear();
 
 #ifdef TORRENT_WINDOWS
+		win_path_string file_path = convert_to_win_path_string(inf);
 		// windows does not allow trailing / or \ in
 		// the path when removing files
-		std::string pruned;
-		if (inf[inf.size() - 1] == '/'
-			|| inf[inf.size() - 1] == '\\')
-			pruned = inf.substr(0, inf.size() - 1);
-		else
-			pruned = inf;
+		while (!file_path.empty() && (
+			file_path.back() == '/' ||
```

Comment:
when ``win_path_string`` is ``std::wstring``, I would expect this comparison, of ``wchar_t`` and ``char`` to yield a warning. no?